packages/core/useDevicePlatform/index.ts

```diff
@@ -0,0 +1,96 @@
+import { readonly, ref, watch } from 'vue'
+import { type ConfigurableWindow, defaultWindow } from '../_configurable'
+
+export type OperatingSystem =
+  | 'iOS'
+  | 'Android OS'
+  | 'BlackBerry OS'
+  | 'Windows Mobile'
+  | 'Amazon OS'
+  | 'Windows 3.11'
+  | 'Windows 95'
+  | 'Windows 98'
+  | 'Windows 2000'
+  | 'Windows XP'
+  | 'Windows Server 2003'
+  | 'Windows Vista'
+  | 'Windows 7'
+  | 'Windows 8'
+  | 'Windows 8.1'
+  | 'Windows 10'
+  | 'Windows ME'
+  | 'Windows CE'
+  | 'Windows 11'
+  | 'Open BSD'
+  | 'Sun OS'
+  | 'Linux'
+  | 'Mac OS'
+  | 'QNX'
+  | 'BeOS'
+  | 'OS/2'
+  | 'Chrome OS'
+
+type OperatingSystemRule = [OperatingSystem, RegExp]
+
+const operatingSystemRules: OperatingSystemRule[] = [
+  ['iOS', /iP(hone|od|ad)/],
+  ['Android OS', /Android/],
+  ['BlackBerry OS', /BlackBerry|BB10/],
+  ['Windows Mobile', /IEMobile/],
+  ['Amazon OS', /Kindle/],
+  ['Windows 3.11', /Win16/],
+  ['Windows 95', /(Windows 95)|(Win95)|(Windows_95)/],
+  ['Windows 98', /(Windows 98)|(Win98)/],
+  ['Windows 2000', /(Windows NT 5.0)|(Windows 2000)/],
+  ['Windows XP', /(Windows NT 5.1)|(Windows XP)/],
+  ['Windows Server 2003', /(Windows NT 5.2)/],
+  ['Windows Vista', /(Windows NT 6.0)/],
+  ['Windows 7', /(Windows NT 6.1)/],
+  ['Windows 8', /(Windows NT 6.2)/],
+  ['Windows 8.1', /(Windows NT 6.3)/],
+  ['Windows 10', /(Windows NT 10.0)/],
+  ['Windows ME', /Windows ME/],
+  ['Windows CE', /Windows CE|WinCE|Microsoft Pocket Internet Explorer/],
+  ['Open BSD', /OpenBSD/],
+  ['Sun OS', /SunOS/],
+  ['Chrome OS', /CrOS/],
+  ['Linux', /(Linux)|(X11)/],
+  ['Mac OS', /(Mac_PowerPC)|(Macintosh)/],
+  ['QNX', /QNX/],
+  ['BeOS', /BeOS/],
+  ['OS/2', /OS\/2/],
+]
+
+function _detectOS(ua: string): OperatingSystem | null {
+  for (let i = 0, count = operatingSystemRules.length; i < count; i++) {
+    const [os, regex] = operatingSystemRules[i]
+    const match = regex.exec(ua)
+
+    if (match)
+      return os
+  }
+
+  return null
+}
+
+export function useDevicePlatform(options: ConfigurableWindow = {}) {
+  const { window = defaultWindow } = options
+
+  const os = ref<OperatingSystem | null>(null)
+
+  if (window) {
+    watch(
+      () => window.navigator.userAgent,
+      () => {
```

Comment:
can `window.navigator.userAgent` be reactive?

Reply:
We were actually not sure about this, the problem is that it is not “reactive” either way. The user agent cannot change just like that.

However, the watcher through the watch immediate ensures that the value can be read directly at the beginning

Reply:
Doesn't changing the emulated device in the DevTools update the User-Agent ?
![Screenshot 2025-03-31 at 11 47 27 AM](https://github.com/user-attachments/assets/1a20ad94-8d6c-41f0-8284-6765313b9d01)

Reply:
`window.navigator.userAgent` isnt a signal/ref (or is it?), so the watcher cant detect the changes.

Reply:
![image](https://github.com/user-attachments/assets/f265f214-036e-433c-a9fa-68054c37b414)

https://play.vuejs.org/#eNp9UctOwzAQ/BXLF1IpckGFS9VGKqgHOACCHi2hKNm6Lolt+ZFUivLvrB21cEC92TOzuzO7A90Yw7oAdElXrrLSeOLAB1NwJVujrScD6UtfHXJS6dYEDzUZyd7qltxg2Q1XXCU+y2bropeq1j1TZSdF6bVlwYHdCFA+J5mCPj1nZF2QgSuCHZXTDbBGi4zT1IZ4K4UAy2lOLgVcjTkZZNtCLUsPSxQFGBHmajWfXKNf/HhoTYOKYgfOr+aXL82pdzhtLwU7Oq0wbTLAaQwlG7Bvxkt0w+lysha5sml0/5KwODA/49UBqu9/8KM7RYzTdwuYuwMMceZ8aQX4id5+vsIJ3xey1XVoUH2F/ADcVIgeJ9ljUDXa/qNLbp/TzaQSO7c9eVDuHCoajcox6TnF2z1dif5rd8HuUx3eALf41YGNPXGBC/bA7m7p+AMEpMIk

Even tho the userAgent changed (see console), the watch will never detect it. Unless we can get a browser event to detect it 😅